### PR TITLE
New branch

### PR DESCRIPTION
**Pull request by VitaliyProgrammer**

### CODE REVIEW

**Review by okuzan — CHANGES_REQUESTED** [review]
Make sure your build passes locally before creating a PR - `mvn clean verify`

File: src/main/java/core/basesyntax/Area.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface Area {
+    double area();
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/Area.java:4]: better naming
```suggestion
public interface AreaCalculator {
    double calculateArea();
```

File: src/main/java/core/basesyntax/Circle.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    private final DecimalFormat decimalFormat = new DecimalFormat("#.##");
+
+    {
+        decimalFormat.setRoundingMode(RoundingMode.HALF_UP);
+    }
+
+    public Circle(Color color, int radius) {
+        super(color);
+        this.radius = radius;
+    }
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/Circle.java:17]:
```suggestion
    public Circle(Color color, int radius) {
        super(color);
        this.radius = radius;
    }
```

File: src/main/java/core/basesyntax/Circle.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+
+public class Circle extends Figure {
+    private final double radius;
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/Circle.java:7]: radius expected to be int
```suggestion
    private final int radius;
```

File: src/main/java/core/basesyntax/Circle.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    private final DecimalFormat decimalFormat = new DecimalFormat("#.##");
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/Circle.java:8]: we dont need that as radius is `int`
```suggestion
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int type = (int) (Math.random() * 5);
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/FigureSupplier.java:7]: magic number

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+                int base = (int) (Math.random() * 100 + 1);
+                int height = (int) (Math.random() * 100 + 1);
+                return new RightTriangle(color, base, height);
+            case 4:
+                int longBase = (int) (Math.random() * 100 + 1);
+                int shortBase = (int) (Math.random() * 100 + 1);
+                int heightBase = (int) ((Math.random() * 100) + 1);
+                return new IsoscelesTrapezoid(color, longBase, shortBase, heightBase);
+            default:
+                return null;
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/FigureSupplier.java:31]: never return null
```suggestion
                return getDefaultFigure();
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        }
+    }
+
+    public Figure getDefaultFigure() {
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/FigureSupplier.java:35]: any reason to make it public?
```suggestion
    private Figure getDefaultFigure() {
```

File: src/main/java/core/basesyntax/RightTriangle.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+    @Override
+    public double area() {
+        return (base * height) / 2.0;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Shape: square, area: " + area() + "sq. units, side: " + base
+                + "sq. units, height: " + "sq. color: " + getColor());
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/RightTriangle.java:21]:
```suggestion
                + "sq. units, height: " + height + "sq. color: " + getColor());
```

File: src/main/java/core/basesyntax/RightTriangle.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    }
+
+    @Override
+    public double area() {
+        return (base * height) / 2.0;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Shape: square, area: " + area() + "sq. units, side: " + base
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/RightTriangle.java:20]:
```suggestion
        System.out.println("Shape: triangle, area: " + area() + "sq. units, side: " + base
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int type = (int) (Math.random() * 5);
+        Color color = colorSupplier.getRandomColor();
+
+        switch (type) {
+            case 0:
+                int radius = (int) (Math.random() * 100 + 1);
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/FigureSupplier.java:12]: create `private final Random random` class-level field and use it
```suggestion
                int radius = random.nextInt(MAX_VALUE);
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+                int width = (int) (Math.random() * 100 + 1);
+                return new Rectangle(color, length, width);
+            case 3:
+                int base = (int) (Math.random() * 100 + 1);
+                int height = (int) (Math.random() * 100 + 1);
+                return new RightTriangle(color, base, height);
+            case 4:
+                int longBase = (int) (Math.random() * 100 + 1);
+                int shortBase = (int) (Math.random() * 100 + 1);
+                int heightBase = (int) ((Math.random() * 100) + 1);
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/FigureSupplier.java:28]: rewrite this code accordingly to the prev comment

File: src/main/java/core/basesyntax/IsoscelesTrapezoid.java [hunk 12]
```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private final int longBase;
+    private final int shortBase;
+
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/IsoscelesTrapezoid.java:6]:
```suggestion
```

File: src/main/java/core/basesyntax/Main.java [hunk 13]
```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[5];
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/Main.java:6]: magic number

File: src/main/java/core/basesyntax/Main.java [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+
+        for (int i = 0; i < figures.length; i++) {
+            figures[i] = figureSupplier.getRandomFigure();
+        }
+
+        for (Figure figure : figures) {
+            figure.draw();
+        }
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/Main.java:14]: not implemented:
The first half of figures in this array should be generated with random parameters.
The other half of the figures should have the same, default parameters.

**Review by kerrrusha — CHANGES_REQUESTED** [review]

File: git reset 758e6c4 [hunk 15]
```diff
@@ -0,0 +1,38 @@
+[33m758e6c4[m[33m ([m[1;36mHEAD -> [m[1;32mnew_branch[m[33m, [m[1;31morigin/new_branch[m[33m)[m FixedFigures2
+[33mbbc55ed[m FixedFigures
```
Comment (sokimaaa) [inline git reset 758e6c4:2]: what is it? remove

File: src/main/java/core/basesyntax/Circle.java [hunk 16]
```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+
+public class Circle extends Figure {
+    private static final DecimalFormat decimalFormat = new DecimalFormat("#.##");
```
Comment (sokimaaa) [inline src/main/java/core/basesyntax/Circle.java:7]: java code convention violations
```suggestion
    private static final DecimalFormat decimalFormat = new DecimalFormat("#.##");
```

Comment (sokimaaa) [inline src/main/java/core/basesyntax/Circle.java:7]: any usage of this?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 17]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private final double length;
+    private final double width;
+
+    public Rectangle(Color color, double length, double width) {
+
```
Comment (sokimaaa) [inline src/main/java/core/basesyntax/Rectangle.java:8]:
```suggestion
```

File: src/main/java/core/basesyntax/RightTriangle.java (hunk 8 above)
Comment (sokimaaa) [reply src/main/java/core/basesyntax/RightTriangle.java:21]: not fixed prev comment:  "height: " + height

**Review by sokimaaa — CHANGES_REQUESTED** [review]

**Comment by VitaliyProgrammer** [comment]
Please check my FixedFigure3..

**Review by sokimaaa — APPROVED** [review]

**Pull request closed without merging**